misc/testerina/modules/testerina-core/src/main/java/org/ballerinalang/testerina/natives/test/FunctionMock.java

```diff
@@ -92,26 +93,93 @@ private static Object callFunction(String returnVal, ArrayValue args) {
                 methodName, argsList.toArray());
     }
 
-    private static String getClassName(String methodName, String orgName, String packageName, String version)
-            throws IOException, ClassNotFoundException {
+    private static String getClassName(String mockMethodName, String orgName, String packageName, String version,
```

Comment:
Is there a possibility to break down some child methods, since this method is very long.

Reply:
I separated the validation part into a different function called `validateFunctionSignature`